### PR TITLE
Multi predictions trainer

### PR DESCRIPTION
**Pull request by sgugger**
This allows the `Trainer` to properly return predictions when the model has several outputs (for instance, all models `XxxForMultipleChoice`). This should unlock progress in #7032 where the start and end logits are both required.

### CODE REVIEW

**Review by LysandreJik — COMMENTED** [review]
Won't this fail if the model has `output_attentions` or `output_hidden_states` set to `True` in their configuration? the `logits` variable would be made of every output, not only the first one. If these are tuples they can't be used with `.detach`, for one, but it wouldn't even make sense to take them into account for the loss computation.

This works only if the model outputs values that should only be used to compute the loss, right?

**Comment by sgugger** [comment]
Yes this doesn't work if the model has `output_attentions` or `output_hiddens` in the config set to True: it needs the outputs to be a tuple of tensor with optional loss at the beginning. Documenting this is next when I make a template/protocol for models that `Trainer` supports. I can add a clean error message if one output is detected to be a tuple of tensors and add asserts that `output_attention` and `output_all_hiddens` are False in the config (if those arguments can be found).

**Review by LysandreJik — APPROVED** [review]
Okay, that sounds good to me. I guess users could override the `prediction_step` anyway if they didn't want to take the full range of inputs.

**Comment by julien-c** [comment]
@sgugger are you saying that the eval in Trainer-backed https://github.com/huggingface/transformers/blob/master/examples/multiple-choice/run_multiple_choice.py is not currently working?

**Comment by LysandreJik** [comment]
I'm pretty sure he meant `XxxForQuestionAnswering`, not multiple choice

**Comment by julien-c** [comment]
> I'm pretty sure he meant `XxxForQuestionAnswering`, not multiple choice

Oh yes, makes sense now. Thanks @LysandreJik ;)

**Comment by sgugger** [comment]
Yes @LysandreJik reads my thoughts right, sorry about the typo ;-)

**Review by LysandreJik — APPROVED** [review]
Nice assertions.
